routing/pathfind.go

```diff
@@ -784,19 +785,12 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 			return
 		}
 
-		// netAmountToReceive is the amount that the node that is added
-		// to the distance map needs to receive from a (to be found)
-		// previous node in the route. The inbound fee of the receiving
-		// node is already subtracted from this value. The previous node
-		// will need to pay the amount that this node forwards plus the
-		// fee it charges plus this node's inbound fee.
-		netAmountToReceive := amountToSend +
-			lnwire.MilliSatoshi(outboundFee)
-
 		// Check if accumulated fees would exceed fee limit when this
 		// node would be added to the path.
-		totalFee := int64(netAmountToReceive) - int64(amt)
-		if totalFee > 0 && lnwire.MilliSatoshi(totalFee) > r.FeeLimit {
+		totalFee := toNodeDist.outboundFee +
+			lnwire.MilliSatoshi(inboundFee)
+
+		if lnwire.MilliSatoshi(totalFee) > r.FeeLimit {
```

Comment:
I think further up, it is already enforced that `in > -out`. These casts don't really matter I think, because on the lowest level these data types are the same?

Reply:
This checks only the hop fee, not the total fee accrued over the path, but the fee limit is for the whole path, I think.